pkg/appgateserver/endpoint_selector.go

```diff
@@ -15,28 +15,29 @@ import (
 // optimizations (i.e. quality of service implementations) are left as an exercise
 // to gateways.
 func (app *appGateServer) getRelayerUrl(
-	serviceId string,
 	rpcType sharedtypes.RPCType,
-	supplierEndpoints []*sdktypes.SingleSupplierEndpoint,
+	sessionEndpoints shannonsdk.FilteredSession,
 	requestUrlStr string,
-) (supplierEndpoint *sdktypes.SingleSupplierEndpoint, err error) {
-	// Filter out the supplier endpoints that match the requested serviceId.
-	validSupplierEndpoints := make([]*sdktypes.SingleSupplierEndpoint, 0, len(supplierEndpoints))
+) (supplierEndpoint shannonsdk.Endpoint, err error) {
+	endpoints, err := sessionEndpoints.AllEndpoints()
```

Comment:
```
	// AppGateServer uses the custom getRelayerUrl instead of leveraging the SDK's
	// filter to select the next endpoint to use.
	// This is because it needs to maintain the state of the last selected endpoint
	// and have a view on the original request URL to determine the next endpoint.
	// This behavior is specific to the AppGateServer and needed by clients that
	// need to instrument the endpoint selection strategy, such as the Load testing tool.
```